src/components/calcite-input-time-picker/calcite-input-time-picker.tsx

```diff
@@ -0,0 +1,276 @@
+  scoped: true
+})
+export class CalciteInputTimePicker {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteInputTimePickerElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** The disabled state of the time input */
+  @Prop({ reflect: true }) disabled?: boolean = false;
+
+  /** Format of the hour value (12-hour or 24-hour) (this will be replaced by locale eventually) */
+  @Prop({ reflect: true }) hourDisplayFormat: "12" | "24" = "12";
+
+  /** The name of the time input */
+  @Prop({ reflect: true }) name?: string;
+
+  /** The scale (size) of the time input */
+  @Prop({ reflect: true }) scale: "s" | "m" | "l" = "m";
+
+  /** number that specifies the granularity that the value must adhere to */
+  @Prop({ reflect: true }) step = 60;
+
+  /** The color theme of the time-picker */
+  @Prop({ reflect: true }) theme: Theme;
+
+  /** The selected time */
+  @Prop({ reflect: true, mutable: true }) value?: string;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  //--------------------------------------------------------------------------
+
+  private calciteInputEl: HTMLCalciteInputElement;
+
+  private referenceElementId: string = guid();
+
+  //--------------------------------------------------------------------------
+  //
+  //  State
+  //
+  //--------------------------------------------------------------------------
+
+  @State() open = false;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * Fires when the time value has changed.
+   */
+  @Event() calciteInputTimePickerChange: EventEmitter<string>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  private inputBlurHandler = (): void => {
+    this.open = false;
+    const newValue = validateTimeString(this.calciteInputEl.value);
+    if (newValue) {
+      this.setValue(newValue);
+    } else {
+      this.setValue(validateTimeString(this.value));
+    }
+  };
+
+  private inputFocusHandler = (): void => {
+    this.open = true;
+  };
+
+  private inputInputHandler = (event: CustomEvent): void => {
+    const value = event.detail.value;
+    const validatedValue = validateTimeString(value);
+    if (validatedValue) {
+      debugger;
+      this.setValue(validatedValue);
+    } else if (!value) {
+      debugger;
+      this.setValue(value);
+    }
+  };
+
+  @Listen("keydown")
```

Comment:
nitpick: Its preferred to use `onKeyDown` on the VNode instead of listeners`. Generally, the `@Listen` is used for window/body events.